main.py

```diff
@@ -115,26 +115,30 @@ def process(data, factors):
             average[factor_set][surface] = np.average(np.array(average[factor_set][surface]), axis=0)
     return average
 
-def sort_directions(data):
+def choose_direction(data, average):
     user_surfaces = []
     for direction_name, surface_arrays in data.items():
-        user_surfaces.append((direction_name, len(surface_arrays.keys()),surface_arrays.keys()))
-    return sorted(user_surfaces, key=lambda x: x[1], reverse=True)
-
+        user_surfaces.append((direction_name, len(surface_arrays.keys())))
+    #get the number of surfaces stored in average as keys
+    user_surfaces = sorted(user_surfaces, key=lambda x: x[1], reverse=True)
+    chosen_surface_num = user_surfaces[0][1]
+    average_surface_num = len(average[next(iter(average))].keys())
```

Comment:
@harrywhoforks what's the purpose of ```average[next(iter(average))]``` here?

Reply:
the 'average' dictionary stores two keys 'annual' and 'summer'. so I was thinking next(iter(average)) to get the first key, which is either annual or summer, and then average['annual'].keys to get all keys, which are the user_surfaces, and then len() to get the number of user_surfaces.

Reply:
@harrywhoforks by calling iter() you are creating a iterator which iterates the dictionary, and next() will look for the iterator's current returned value (which is None), so that it returns the first iterated value to you.

Reply:
@harrywhoforks As you can see the next(iter()) thing creates extra overhead both in terms of memory and space. Maybe a better method would be storing the value back when we're processing the average?

Reply:
what about i use average[average.keys()[0]]?

Reply:
```average_surface_num``` is removed in 46d9a6ba1b1df05f123a3ca0479fb4e7e9080320, resolving this discussion.